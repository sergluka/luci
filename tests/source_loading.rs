use insta::assert_debug_snapshot;
use luci::execution::SourceCodeLoader;
use test_case::test_case;

#[test_case("00", "tests/source_loading/00-the-simplest-case.luci.yaml", &["."])]
#[test_case("01", "tests/source_loading/01-one-inclusion.luci.yaml", &["."])]
#[test_case("00.a", "00-the-simplest-case.luci.yaml", &["tests/source_loading"])]
#[test_case("01.a", "01-one-inclusion.luci.yaml", &["tests/source_loading"])]
#[test_case("02", "tests/source_loading/02-direct-cyclic-inclusion.luci.yaml", &["."])]
#[test_case("02.a", "02-direct-cyclic-inclusion.luci.yaml", &["tests/source_loading"])]
#[test_case("03", "03-indirect-cyclic-inclusion.luci.yaml", &["tests/source_loading"])]
#[test_case("04", "04-diamond.luci.yaml", &["tests/source_loading", "tests/source_loading/04-diamond"])]
fn load_sources(name: &str, main: &str, search_paths: &[&str]) {
    let mut loader = SourceCodeLoader::new();
    loader.search_path = search_paths.into_iter().copied().map(From::from).collect();
    let outcome = loader.load(main);
    assert_debug_snapshot!(name, outcome);
}
