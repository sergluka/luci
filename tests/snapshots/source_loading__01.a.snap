---
source: tests/source_loading.rs
expression: outcome
---
Ok(
    (
        KeyScenario(
            1v1,
        ),
        {
            "tests/source_loading/00-the-simplest-case.luci.yaml": Source {
                source_file: "tests/source_loading/00-the-simplest-case.luci.yaml",
                subs: {},
                scenario: Scenario {
                    types: [],
                    subroutines: [],
                    actors: [],
                    dummies: [],
                    events: [],
                    no_extra: NoExtra,
                },
            },
            "tests/source_loading/01-one-inclusion.luci.yaml": Source {
                source_file: "tests/source_loading/01-one-inclusion.luci.yaml",
                subs: {
                    SubroutineName(
                        "f00",
                    ),
                },
                scenario: Scenario {
                    types: [],
                    subroutines: [
                        DefDeclareSub {
                            file_name: "00-the-simplest-case.luci.yaml",
                            subroutine_name: SubroutineName(
                                "f00",
                            ),
                            no_extra: NoExtra,
                        },
                    ],
                    actors: [],
                    dummies: [],
                    events: [],
                    no_extra: NoExtra,
                },
            },
        },
    ),
)
